test/test_genome.py

```diff
@@ -346,89 +347,118 @@ def test_is_gene_in_output_region(rng_seed):
     assert not genome._is_gene_in_output_region(11)
 
 
-def test_mutate_hidden_region(rng_seed):
+def test_mutation_rate(rng_seed, mutation_rate):
+    n_input = 1
+    n_output = 1
+    n_column = 4
+    n_row = 3
+    genome = cgp.Genome(n_input, n_output, n_column, n_row, None, (cgp.Add, cgp.Sub))
     rng = np.random.RandomState(rng_seed)
-    genome = cgp.Genome(1, 1, 3, 1, None, (cgp.Add, cgp.ConstantFloat))
-    dna = [
-        ID_INPUT_NODE,
-        ID_NON_CODING_GENE,
-        ID_NON_CODING_GENE,
-        1,
-        0,
-        0,
-        1,
-        0,
-        0,
-        0,
-        0,
-        2,
-        ID_OUTPUT_NODE,
-        3,
-        ID_NON_CODING_GENE,
-    ]
-    genome.dna = list(dna)
-    active_regions = cgp.CartesianGraph(genome).determine_active_regions()
+    genome.randomize(rng)
 
-    # mutating any gene in inactive region returns True
-    assert genome._mutate_hidden_region(list(dna), 3, active_regions, rng) is True
-    assert genome._mutate_hidden_region(list(dna), 4, active_regions, rng) is True
-    assert genome._mutate_hidden_region(list(dna), 5, active_regions, rng) is True
+    def count_n_immutable_genes(n_input, n_output, n_row):
+        counter = n_input * 3  # 3 is the arity
+        counter += n_output * 2  # 2 of 3 output genes can not be mutated
+        if n_input == 1:
```

Comment:
what happens for `n_input != 1`?

Reply:
`counter` remains on the previous value. If I am not mistaking only in the case of `n_input = 1` the input gene addresses can not be mutated (for hidden genes in the first column). If `n_input >= 2` they have more than one permissible value and are therefore mutable.

Reply:
i see. well, i think your test is anyway adapted to this specific configuration of the genome, isn't it? for example would it be robust against increasing the number of columns? if it's specific, you don't need the `if` since you define `n_inputs =1` before.

Reply:
Changing `n_columns` wouldn't matter. Only in the first column the input gene of hidden can not be mutated (in the case of `n_input = 1`). Thats why n_rows matters. (Unless I am missing some other case for which only one 1 allele value is permitted).  The test would still pass with `n_input != 1` but it would loose some capacity.